core/controllers/access_validators_test.py

```diff
@@ -663,13 +663,13 @@ def test_any_user_can_access_review_tests_page(self) -> None:
     def test_no_user_can_access_unpublished_story_review_sessions_page(
         self
     ) -> None:
-        self.get_html_response(
+        self.get_json(
```

Comment:
This change seems fine. But some notes:

- In the handler above why are we still using get_html_response()?
- If this pattern is incorrect elsewhere could you please fix it in those places too so that other devs don't propagate this pattern further?

Thanks!

Reply:
Thanks for the review,
The get_html_response() was used in previous tests because everything was handled in the backend but now we can change it to get_json() in all cases but now we are using it only on the ones that return a success code thus they return a html response we can switch all of them to .get_json() on this PR but I think it will be confusing to the next pages we are migrating (the ones that are currently being migrated) so I think we can do it later just to unify all the code at the end of the migration. 
what do you think @seanlip ?

Reply:
@seanlip PTAL

Reply:
@Ahmed-Bhouri Sorry, I'm somewhat confused. Why does get_json() return a html response in the case of success? In general each handler should return either a html or json response and be consistent about it. Could you please clarify this?

I think that, once we figure out the correct approach, it would be nice to make that fix throughout so that other devs don't copy/paste it incorrectly. Doing it here is fine if this PR is close to being merged, but you can also open a separate PR for that concurrently if you like.

Reply:
@Ahmed-Bhouri I think what happens is that each server endpoint returns a specific thing. So it will depend on which endpoint you are testing -- you can test both endpoints if you like, and the test should make clear whether we are testing the HTML or the JSON endpoint.

Does that help in any way?

Reply:
@seanlip sorry for my late reply I had a busy week.
Well after further investigation I had some concepts mixed up. So the tests where the server returns a success code 200 It also had to respond with a text/html. Thus, we should use the get_html_response() otherwise we use the get_json(). I have checked the rest of the tests and all of them use the appropriate function.
Therefore this requires no more changes.

Reply:
@Ahmed-Bhouri I am sorry, I still don't understand what you're talking about here.

A handler is either an HTML handler or a JSON handler. Both HTML and JSON handlers can return success or failure codes. It does not make sense for the handler type to be dependent on whether the response succeeds or fails, but this is what your comment implies.

Could you please elaborate more on what is going on here?

Reply:
@seanlip 
when testing the access to a page If we are testing a successful access the response will be a HTML response if we are testing for a fail in most cases the expected behavior is a redirect along with a status code indicating the reason of the redirect in that case the handler will no longer return a HTML response It will return a JSON response.

Reply:
@Ahmed-Bhouri Not quite. When you write backend tests, you are testing a handler (GET/POST/PUT/DELETE with a URL path), and you need to test both the success and failure cases.

What handler are you testing? Are you saying the same handler returns HTML in one case and a HTML redirect in the other that leads to a JSON handler? If the latter, that's odd, HTML handlers should redirect to HTML handlers. Can you specify the specific handler you are talking about in future responses?

Reply:
@seanlip we are testing for get handlers that return a html response on success but will return a redirect on fail. in the case of a redirect we dont test what the redirect link returns we just check the json of the redirect to know weather the behavior is correct or not.

Reply:
@Ahmed-Bhouri But then I would expect this to be doing a "get HTML", seeing that the request has status int 302, and then examining response.headers['location'] ... right? See e.g. the tests for redirction in core/controllers/acl_decorators_test.py.